File: lengthOfLongestSubstring.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int aPointer=0;
        int bPointer=0;
        int max=0;
        HashSet<Character> hashset = new HashSet();
        
        while (bPointer<s.length()){
         if(!hashset.contains(s.charAt(bPointer))){
             hashset.add(s.charAt(bPointer));
             bPointer++;
             max =Math.max(hashset.size(),max);
         } else {
                hashset.remove(s.charAt(aPointer));
                aPointer++;
            }
        }
        return max;
    }
}

File: lengthOfLongestSubstring.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n = s.size();
        int l ,r,i=0;
        map<char, int> mp;
        int curLength=0;
        int result=0;
        while(i<n){
            mp[s[i]]++;
            curLength++;
            while(mp[s[i]] >1) {
                mp[s[l]]--;
                l++;
                curLength--;
            }
            r++;
            i++;
            result = max(result , curLength);
        }
        return result;
        
    }
};
